etckeeper.8
.\" -*- nroff -*-
.TH ETCKEEPER 8 "" "" ""
.SH NAME
etckeeper \- store /etc in git, mercurial, bazaar, or darcs
.SH SYNOPSIS
.B etckeeper command [-d directory]
.SH DESCRIPTION
etckeeper manages /etc be stored in a git, mercurial, bazaar, or darcs
repository. By default each of the commands operates on /etc, but a
different directory can be specified to operate on a clone of the /etc
repository located elsewhere.
.SH COMMANDS
.TP
.B init
This initialises and sets up a git, mercurial, bazaar, or darcs
repository (depending on the VCS setting in
/etc/etckeeper/etckeeper.conf). Typically this is run in /etc once
when starting to use etckeeper on a machine. It can also be used to
initialise a clone of the /etc repository located elsewhere.
.TP
.B commit [message]
Commits all changes in /etc to the repository. A commit message can be
specified. You may also use the underlying VCS to commit manually.
(Note that etckeeper commit will notice if a user has used sudo or su to
become root, and record the original username in the commit.)
.TP
.B pre-commit
This is called as a pre-commit hook. It stores metadata and does sanity
checks.
.TP
.B pre-install
This is called by apt's DPkg::Pre-Install-Pkgs hook, or by equivalent hooks
of other package managers. It allows committing any uncommitted changes before
packages are installed, upgraded, etc.
.TP
.B post-install
This is called by apt's DPkg::Post-Invoke hook, or by equivalent hooks
of other package managers. It commits changes made by packages into the
repository. (You can also call this by hand after running dpkg by hand.)
.TP
.B unclean
This returns true if the directory contains uncommitted changes.
.TP
.B update-ignore [-a]
This updates the VCS ignore file. Content outside a "managed by etckeeper"
block is not touched. This is generally run when upgrading to a new version
of etckeeper. (The -a switch will add a "managed by etckeeper" block if
one is not present.)
.TP
.B vcs subcommand [options ...]
You can use this to run any subcommand of the VCS that etckeeper is
configured to run. It will be run in /etc. For example, "etckeeper vcs
diff" will run "git diff", etc.
.TP
.B uninit [-f]
This command DESTROYS DATA! It is the inverse of the init command, removing
VCS information and etckeeper's own bookkeeping information from the
directory. Use with caution. A typical use case would be to run etckeeper
uninit, then modify etckeeper.conf to use a different VCS, and then run
etckeeper init. (The -f switch can be used to force uninit without
prompting.)
.SH FILES
/etc/etckeeper/etckeeper.conf is the configuration file.

/etc/etckeeper also contains directories containing the programs that are
run for each of the above commands.
.SH ENVIRONMENT VARIABLES
ETCKEEPER_CONF_DIR path to configuration directory instead of default /etc/etckeeper.
.SH SEE ALSO
/usr/share/doc/etckeeper/README.md.gz
.SH AUTHOR 
Joey Hess <id@joeyh.name>


debian/changelog
etckeeper (1.17) unstable; urgency=medium

  * Fix name of DNF plugin.
  * Add --version
    Thanks Andreas Wansner.
  * New website, http://etckeeper.branchable.com/
  * Add build-depends on dh-python.

 -- Joey Hess <id@joeyh.name>  Mon, 22 Dec 2014 16:42:26 -0400

etckeeper (1.16) unstable; urgency=medium

  * Added support for Fedora's DNF highlevel package manager.
    Thanks, Peter Listiak and Petr Spacek.
  * Add architecture info to dpkg list-installed. Closes: #768145
  * Orphaned the Debian package.

 -- Joey Hess <joeyh@debian.org>  Fri, 07 Nov 2014 21:30:50 -0400

etckeeper (1.15) unstable; urgency=medium

  * Recommend cron-daemon, rather than cron, as etckeeper only needs
    cron.daily functionality. Closes: #762721

 -- Joey Hess <joeyh@debian.org>  Thu, 16 Oct 2014 12:21:48 -0400

etckeeper (1.14) unstable; urgency=medium

  * Handle failure to commit in post-install, pre-install by showing a
    warning, rather than propigating the error to apt.
    This avoids breaking the apt run when eg, git is misconfigured and
    cannot commit.
    pre-install already did this when it was able to use debconf to display a
    message, but now debconf is not used, and it always behaves this way.
    Closes: #760011

 -- Joey Hess <joeyh@debian.org>  Thu, 04 Sep 2014 15:47:32 -0400

etckeeper (1.13) unstable; urgency=medium

  * Ignore check-mk-agent-logwatch's FHS violating
    /etc/check_mk/logwatch.state. Closes: #753903
  * Only allow [-a-z_] in etckeeper commands to avoid any possible directory
    traversal etc issues.
  * update-ignore, uninit: Fix parsing of ignore files containing '\'

 -- Joey Hess <joeyh@debian.org>  Sat, 09 Aug 2014 13:05:55 -0400

etckeeper (1.12) unstable; urgency=medium

  * Portability fixes. Thanks, Harald Dunkel.
  * Add support for pushing to multiple remote repositories.
    Thanks, Rouben.
  * Fix handling of git ignores like dir/*
    Thanks, Pim van den Berg

 -- Joey Hess <joeyh@debian.org>  Fri, 13 Jun 2014 12:03:06 -0400

etckeeper (1.11) unstable; urgency=low

  * Fix too broad matching of .gitignored files.
    Closes: #732339

 -- Joey Hess <joeyh@debian.org>  Tue, 17 Dec 2013 22:24:29 -0400

etckeeper (1.10) unstable; urgency=low

  * Remove lvm/backup from default ignores, because lvm
    documentation recommends backing that up, for use by
    vgcfgrestore.
  * Fix exporting of some git variables. Closes: #728583

 -- Joey Hess <joeyh@debian.org>  Sun, 03 Nov 2013 12:14:02 -0400

etckeeper (1.9) unstable; urgency=low

  * Fix git update-ignore syntax. Closes: #721873

 -- Joey Hess <joeyh@debian.org>  Wed, 04 Sep 2013 21:43:42 -0400

etckeeper (1.8) unstable; urgency=low

  * Avoid listing .gitignored files in .etckeeper file. Closes: #607665
    Thanks, Zdenek Crha

 -- Joey Hess <joeyh@debian.org>  Wed, 04 Sep 2013 09:31:37 -0400

etckeeper (1.7) unstable; urgency=low

  * Fix hilarious typo hardcoding my name. Closes: #718425

 -- Joey Hess <joeyh@debian.org>  Wed, 31 Jul 2013 11:33:45 -0400

etckeeper (1.6) unstable; urgency=low

  * Guard git config calls. Closes: #717957

 -- Joey Hess <joeyh@debian.org>  Sat, 27 Jul 2013 12:24:30 -0400

etckeeper (1.5) unstable; urgency=low

  * Quote user and group names, in case one contains a space.
  * Added support for the pacman package manager.
    (Thanks, Tiago Stürmer Daitx)
  * Use user.name and user.email from the .gitconfig file belonging to the
    user who sued or sudoed to root, in preference to making up values for
    that user.
  * cron.daily: Fix typo in stale lockfile handling code.
    Closes: #717908

 -- Joey Hess <joeyh@debian.org>  Fri, 26 Jul 2013 11:03:48 -0400

etckeeper (1.4) unstable; urgency=low

  * Deal with unix^wlinux portability nonsense.

 -- Joey Hess <joeyh@debian.org>  Mon, 17 Jun 2013 12:00:59 -0400

etckeeper (1.3) unstable; urgency=low

  * Fix type -p bashism that crept in via recent patches. Closes: #707319

 -- Joey Hess <joeyh@debian.org>  Wed, 08 May 2013 22:36:30 -0400

etckeeper (1.2) unstable; urgency=low

  * Call type -p in a more compatable way.
  * When a file is owned by a uid or a gid with no corresponding user or
    group, put a numeric chown into .etckeeper. Previously, a broken
    chown was outputted.

 -- Joey Hess <joeyh@debian.org>  Wed, 08 May 2013 11:51:51 -0400

etckeeper (1.1) unstable; urgency=low

  * Fix warning when PUSH_REMOTE is not set. Closes: #706917

 -- Joey Hess <joeyh@debian.org>  Mon, 06 May 2013 09:49:24 -0400

etckeeper (1.0) unstable; urgency=low

  [ Joey Hess ]
  * Unset GIT_DIR and GIT_WORK_TREE. Closes: #689101
  * PUSH_REMOTE can be set to automatically push to a remote on commit.
    Thanks, L. Alberto Giménez
  * Ignore fake-hwclock.data. Closes: #701491

  [ Jelmer Vernooij ]
  * Auto-detect the VCS setting if there already is a repository in
    /etc.

 -- Joey Hess <joeyh@debian.org>  Sat, 04 May 2013 23:45:25 -0400

etckeeper (0.64) unstable; urgency=low

  * Added support for openSUSE's zypper package manager.
    Thanks, Catalin Iacob
  * Add Brazilian Portuguese debconf translation.
    Closes: #685771 Thanks, Adriano Rafael Gomes

 -- Joey Hess <joeyh@debian.org>  Sat, 25 Aug 2012 11:53:08 -0400

etckeeper (0.63) unstable; urgency=low

  * bzr: Improve detection of unclean repos, to work when there are shelved
    changes.
  * uninit: Now preserves parts of the gitignore and similar files that
    are outside the managed by etckeeper block. Closes: #673996
    Thanks, David De La Harpe Golden (Squared Financial)

 -- Joey Hess <joeyh@debian.org>  Sat, 02 Jun 2012 18:22:37 -0400

etckeeper (0.62) unstable; urgency=low

  * Autocommit git staged files. Closes: #662614

 -- Joey Hess <joeyh@debian.org>  Mon, 05 Mar 2012 10:41:46 -0400

etckeeper (0.61) unstable; urgency=low

  * Fix up botched git-rm conffile removal from 0.58.
    The file could be in any of three states; absent, present, or .dpkg-dist.
    Finish fully removing it. Closes: #655836


etckeeper-dnf/etckeeper.py
# etckeeper.py, support etckeeper for dnf
#
# Copyright (C) 2014 Peter Listiak
# https://github.com/plistiak/dnf-etckeeper
#
# Later modifications by Petr Spacek:
# Distutils code below was copied from etckeeper-bzr distributed with v1.15
#

from dnfpluginscore import logger

import os
import dnf


class Etckeeper(dnf.Plugin):

    name = 'etckeeper'

    def _out(self, msg):
        logger.debug('Etckeeper plugin: %s', msg)

    def resolved(self):
        self._out('pre transaction commit')
        command = '%s %s' % ('etckeeper', " pre-install")
        ret = os.system(command)
        if ret != 0:
            raise dnf.exceptions.Error('etckeeper returned %d' % (ret >> 8))

    def transaction(self):
        self._out('post transaction commit')
        command = '%s %s > /dev/null' % ('etckeeper', "post-install")
        os.system(command)

if __name__ == "__main__":
    from distutils.core import setup
    setup(name="dnf-etckeeper",
          packages=["dnf-plugins"],
          package_dir={"dnf-plugins":"etckeeper-dnf"})
